crates/typst-library/src/model/numbering.rs

```diff
@@ -79,7 +79,10 @@ pub fn numbering(
     /// numberings to the `numbering` function without caring whether they are
     /// defined as a pattern or function.
     numbering: Numbering,
-    /// The numbers to apply the numbering to. Must be positive.
+    /// The numbers to apply the numbering to. Must be non-negative.
+    ///
+    /// In general, numbers are counted from one. A number of zero indicates
+    /// that the first element has not yet appeared.
```

Comment:
https://github.com/typst/typst/issues/1896#issuecomment-2005782054 or [headcount – Typst Universe](https://typst.app/universe/package/headcount) depends on this behaviour for equations/images/tables that come before the first heading.